From fa711a15d0eebb70bf93368522f0a051c0c465e8 Mon Sep 17 00:00:00 2001
From: Eric Anholt <eric@anholt.net>
Date: Mon, 5 Feb 2018 18:53:18 +0000
Subject: [PATCH 136/220] drm/vc4: Don't wait for vblank on fkms cursor
 updates.

We don't use the same async update path between fkms and normal kms,
and the normal kms workaround ended up making us wait.  This became a
larger problem in rpi-4.14.y, as the USB HID update rate throttling
got (accidentally?) dropped.

Signed-off-by: Eric Anholt <eric@anholt.net>
---
 drivers/gpu/drm/vc4/vc4_kms.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/vc4/vc4_kms.c b/drivers/gpu/drm/vc4/vc4_kms.c
index 8a411e5..e6448a5 100644
--- a/drivers/gpu/drm/vc4/vc4_kms.c
+++ b/drivers/gpu/drm/vc4/vc4_kms.c
@@ -160,7 +160,8 @@ static u16 vc4_ctm_s31_32_to_s0_9(u64 in)
 	 * display lists before we free it and potentially reallocate
 	 * and overwrite the dlist memory with a new modeset.
 	 */
-	state->legacy_cursor_update = false;
+	if (!vc4->firmware_kms)
+		state->legacy_cursor_update = false;
 
 	drm_atomic_helper_commit_hw_done(state);
 
-- 
1.9.1

